File: pharma-backend/src/routes/classes.js
import { Router } from 'express';
import { pool } from '../db.js';

const router = Router();

router.get('/', async (req, res) => {
  try {
    const { rows } = await pool.query(
      'SELECT id, name, full_name, description FROM drug_classes ORDER BY name'
    );
    res.json(rows);
  } catch (err) {
    console.error(err);
    res.status(500).json({ error: 'Failed to fetch classes' });
  }
});

export default router;

File: pharma-backend/app/routers/classes.py
from fastapi import APIRouter
from app.db import get_connection

router = APIRouter(prefix="/api/classes", tags=["classes"])


@router.get("")
async def list_classes():
    async with get_connection() as conn:
        rows = await conn.fetch("SELECT name, full_name, description FROM drug_classes ORDER BY name")
        return [dict(r) for r in rows]
